ui/src/api/api-fetch.ts

```diff
@@ -326,8 +355,14 @@ async function executeBodyRequestInternal<Output>(
             (response.headers.has('session-expired') ||
                 errorResponse.response.localizationKey === TOKEN_EXPIRED);
 
+        const versionMismatch =
+            response.status === 400 && errorResponse.response.localizationKey === INVALID_VERSION;
+
         if (retryOnTokenExpired && errorIsTokenExpiry) {
             return executeBodyRequestInternal(fetchFunction, false);
+        } else if (versionMismatch) {
+            createDelegates(commonActionCreators).setVersionStatus('reload');
```

Comment:
Redux-storen kautta päätyy main.tsx-filuun tällä tavalla tuo reload-dialogin näyttötoivomus. Huom: edellyttää tässäkin reduxin käyttöä jälleen mistä en pidä.

Reply:
Joo tuo tuntuu hähmäiseltä. Jotenkin ajattelin että tämä menisi toisin päin: backend antaa version ja front tallentaa sen -> joku komponentti x tarkkailee tuota osaa storesta ja jos ne eroaa niin tekee reloadin. Tuokin edellyttää reduxin käyttöä, mutta siellä säilytettäisi tilaa (frontVersion, backendVersion) vs käyttö toiminnon triggerinä (reload). Mutta tämä malli on selvästi parempi siinä että koska backari heittää requestfilteristä, virhe saadaan samanlaisena vaikka versio-mismatchin myötä koko kutsuttua polkua ei olisikaan. 

Eli summa summarum, vaikka tämä näyttää minustakin vähän yöksiltä, se on silti parempi kuin oma ajatukseni, joten :+1:

Reply:
Taitaisi se noinkin päin toimia 🤔. Bäkkärihän voisi palauttaa x-geoviite-backend-version headerin myös virhepalautuksissa kuten 404 eli se toimisi lähes yhtälailla kuin tämäkin.

Tuossahan mitä mietit on lähinnä logiikka eri päin:
* Mun toteutuksella frontti lähettää joka pyynnöllä headerin, jonka bäkkäri tarkistaa ja vastaa tarvittaessa virheellä, jonka frontti käsittelee ja näyttää "vanhentunut versio"-dialogin.
* Sun idiksellä bäkkäri palauttaa versioheaderin joka pyynnöllä, ja ui-koodi tarkistaa mätsääkö se reduksiin tallennettuun, näyttäen mahdollisesti sen "vanhentunut versio"-dialogin.

Ylemmässähän on yksi steppi lisää: frontti joutuu tarkistamaan mitä bäkkäri palautti kummassakin (joko että palautuko eri version headeri vaiko virhevastaus), mutta alemmassa se ei joudu välittämään omaa versiotietoaan bäkkärille eli bäkkäri ei tarkista mitään. Ylemmässä on tosin se hyöty, että siitä jää lokitietoa meille: bäkkäri lokittaa mikäli se palauttaa tuon versiovirheen, kun taas jos frontti toteaa versiovirheen tapahtuneen niin tätä tieto ei jää (meillä ei ole frontilla telemetriaa)